src/control/server/server.go

```diff
@@ -110,10 +114,31 @@ func Start(log *logging.LeveledLogger, cfg *Configuration) error {
 		TargetUser:    runningUser.Username,
 		PCIWhitelist:  strings.Join(cfg.BdevInclude, ","),
 	}
+	log.Debugf("automatic NVMe prepare req: %+v", prepReq)
 	if _, err := bdevProvider.Prepare(prepReq); err != nil {
 		log.Errorf("automatic NVMe prepare failed (check configuration?)\n%s", err)
 	}
 
+	hugePages, err := getHugePageInfo()
+	if err != nil {
+		return errors.Wrap(err, "unable to read system hugepage info")
+	}
+
+	// Don't bother with these checks if there aren't any block devices configured.
+	if cfgHasBdev(cfg) {
+		if hugePages.Free != hugePages.Total {
+			// Not sure if this should be an error, per se, but I think we want to display it
+			// on the console to let the admin know that there might be something that needs
+			// to be cleaned up?
+			log.Errorf("free hugepages does not match total (%d != %d)", hugePages.Free, hugePages.Total)
```

Comment:
I guess I don't totally understand what the consequence of this is. If it isn't hurting the user, I'd be inclined to reduce the error level (do we have warnings? or just info/debug?).

Reply:
To be honest, I don't fully understand all the ramifications either, and I'm not convinced that anyone on the wider team has really internalized this stuff. I wanted to err on the side of being a bit too chatty until we understand things better.

Reply:
Could there be other applications using the hugepages? If so I feel like it would be common to see this.

Reply:
On a dedicated DAOS storage server, probably not. But this way we can test our assumptions. I'd rather be a bit noisy while we're getting more experience with it rather than having mysterious failures. This is a significant change in the way we're using SPDK.

Reply:
OK, works for me.